skrub/_agg_joiner.py

```diff
@@ -0,0 +1,592 @@
+
+        categorical : {"mode", "count", "value_counts"}
+
+        If set to None, ['mean', 'mode'] will be used.
+
+    suffixes : list of str, default=None
+        The suffixes that will be add to each table columns in case of
+        duplicate column names, similar to `("_x", "_y")` in :obj:`~pandas.merge`.
+
+        If set to None, we will use the table index, by looking at their
+        order in 'tables', e.g. for a duplicate columns: price (main table),
+        price_1 (auxiliary table 1), price_2 (auxiliary table 2), etc.
+
+    See Also
+    --------
+    :class:`AggTarget` :
+        Aggregates the target `y` before joining its aggregation
+        on the base dataframe.
+    :class:`FeatureAugmenter` :
+        Augments a main table by automatically joining multiple
+        auxiliary tables on it.
+
+    Examples
+    --------
+    >>> main = pd.DataFrame({
+            "userId": [1, 1, 1, 2, 2, 2],
+            "movieId": [1, 3, 6, 1, 3, 6],
+            "rating": [4.0, 4.0, 4.0, 3.0, 2.0, 4.0],
+            "genre": ["drama", "drama", "comedy", "sf", "comedy", "sf"],
+        })
+    >>> join_agg = AggJoiner(
+            tables=[
+                (main, "userId", ["rating", "genre"]),
+                (main, "movieId", ["rating"]),
+            ],
+            main_key=["userId", "movieId"],
+            suffixes=["_user", "_movie"],
+            agg_ops=["mean", "mode"],
+        )
+    >>> join_agg.fit_transform(main)
+        userId  movieId  rating   genre genre_mode_user  rating_mean_user  rating_mean_movie
+    0       1        1     4.0   drama           drama               4.0   3.5
+    1       1        3     4.0   drama           drama               4.0   3.0
+    2       1        6     4.0  comedy           drama               4.0   4.0
+    3       2        1     3.0      sf              sf               3.0   3.5
+    4       2        3     2.0  comedy              sf               3.0   3.0
+    5       2        6     4.0      sf              sf               3.0   4.0
+    """  # noqa: E501
+
+    def __init__(self, tables, main_key, agg_ops=None, suffixes=None):
+        self.tables = tables
+        self.main_key = main_key
+        self.agg_ops = agg_ops
+        self.suffixes = suffixes
+
+    def fit(self, X, y=None) -> "AggJoiner":
+        """Aggregate auxiliary tables based on the main keys.
+
+        Parameters
+        ----------
+        X : {pandas.Dataframe}
+            Input data, based table on which to left join the
+            auxiliary tables.
+        y : array-like of shape (n_samples), default=None
+            Used to compute the correlation between the generated covariates
+            and the target for screening purposes.
+
+        Returns
+        -------
+        AggJoiner
+            Fitted :class:`AggJoiner` instance (self).
+        """
+        self.check_input(X)
+        agg_px, _ = get_namespace([table for (table, _, _) in self.tables_])
```

Comment:
agg_px feels like a suboptimal name for this object: it implements much more than aggregation (for instance agg_px.aggregate, as below, seems strange)

Reply:
And what does `px` stand for? This needs to be more descriptive.

Reply:
When dealing with namespaces (`np`, `cupy`, `torch`, etc) [scikit-learn uses the generic `xp`](https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/utils/_array_api.py#L358). Since we consider pandas (`pd`) and polars (`pl`), `px` seemed relevant to me.

However, we're not returning either `pd` or `pl` namespace, but our own (either the file `_agg_pandas.py` or `_agg_polars.py`). So `agg_px` was what I came up with.

I'll use `px` to simplify things for now, but feel free to suggest anything else.

Reply:
WDYT?